Formula/octant.rb

```diff
@@ -0,0 +1,42 @@
+class Octant < Formula
+  desc "Kubernetes introspection tool for developers"
+  homepage "https://github.com/vmware/octant"
+  url "https://github.com/vmware/octant.git",
+      :tag      => "v0.6.0",
+      :revision => "a14f6a25a9d1cfa369652f0d237954122f680ca3"
+  head "https://github.com/vmware/octant.git"
+
+  depends_on "go" => :build
+  depends_on "node@10" => :build
+  depends_on "protoc-gen-go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GO111MODULE"] = "on"
+    ENV["GOFLAGS"] = "-mod=vendor"
+
+    dir = buildpath/"src/github.com/vmware/octant"
+    dir.install buildpath.children
+
+    cd "src/github.com/vmware/octant" do
+      system "make", "go-install"
+      ENV.prepend_path "PATH", buildpath/"bin"
+
+      system "make", "web-build"
+      system "make", "generate"
+
+      commit = Utils.popen_read("git rev-parse HEAD").chomp
+      build_time = Utils.popen_read("date -u +'%Y-%m-%dT%H:%M:%SZ' 2> /dev/null").chomp
+      ldflags = ["-X \"main.version=#{version}\"",
+                 "-X \"main.gitCommit=#{commit}\"",
+                 "-X \"main.buildTime=#{build_time}\""]
+
+      system "go", "build", "-o", bin/"octant", "-ldflags", ldflags.join(" "),
+              "-v", "./cmd/octant"
+    end
+  end
+
+  test do
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Tests have been updated